lib/src/channels/http_channel.dart

```diff
@@ -0,0 +1,207 @@
+
+  @override
+  Duration idleTimeout;
+
+  @override
+  int maxConnectionsPerHost;
+
+  @override
+  String userAgent;
+
+  int count = 1;
+
+  LoggingHttpClient({SecurityContext context}) {
+    HttpOverrides.global = null;
+    proxy = HttpClient(context: context);
+    HttpOverrides.global = _httpOverrides;
+  }
+
+  @override
+  set authenticate(
+      Future<bool> Function(Uri url, String scheme, String realm) f) {
+    todo('authenticate');
+    proxy.authenticate = f;
+  }
+
+  @override
+  set authenticateProxy(
+      Future<bool> Function(String host, int port, String scheme, String realm)
+          f) {
+    todo('authenticateProxy');
+    proxy.authenticateProxy = f;
+  }
+
+  @override
+  bool get autoUncompress => proxy.autoUncompress;
+
+  set autoUncompress(bool value) {
+    proxy.autoUncompress = value;
+  }
+
+  @override
+  set badCertificateCallback(
+      bool Function(X509Certificate cert, String host, int port) callback) {
+    todo('badCertificateCallback');
+    proxy.badCertificateCallback = callback;
+  }
+
+  @override
+  set findProxy(String Function(Uri url) f) {
+    todo('findProxy');
+    proxy.findProxy = f;
+  }
+
+  @override
+  void addCredentials(
+      Uri url, String realm, HttpClientCredentials credentials) {
+    todo('addCredentials');
+    proxy.addCredentials(url, realm, credentials);
+  }
+
+  @override
+  void addProxyCredentials(
+      String host, int port, String realm, HttpClientCredentials credentials) {
+    todo('addProxyCredentials');
+    proxy.addProxyCredentials(host, port, realm, credentials);
+  }
+
+  @override
+  void close({bool force = false}) {
+    todo('close');
+    proxy.close(force: force);
+  }
+
+  @override
+  Future<HttpClientRequest> delete(String host, int port, String path) {
+    todo('delete');
+    return proxy.delete(host, port, path);
+  }
+
+  @override
+  Future<HttpClientRequest> deleteUrl(Uri url) {
+    todo('deleteUrl');
+    return proxy.deleteUrl(url);
+  }
+
+  @override
+  Future<HttpClientRequest> get(String host, int port, String path) {
+    _log.log(() => 'getUrl: $host $port $path');
+    return proxy.get(host, port, path);
+  }
+
+  @override
+  Future<HttpClientRequest> getUrl(Uri url) {
+    _log.log(() => 'getUrl: $url');
+    return proxy.getUrl(url);
+  }
+
+  @override
+  Future<HttpClientRequest> head(String host, int port, String path) {
+    todo('head');
+    return proxy.head(host, port, path);
+  }
+
+  @override
+  Future<HttpClientRequest> headUrl(Uri url) {
+    todo('headUrl');
+    return proxy.headUrl(url);
+  }
+
+  @override
+  Future<HttpClientRequest> open(
+      String method, String host, int port, String path) {
+    todo('open');
+    return proxy.open(method, host, port, path);
+  }
+
+  @override
+  Future<HttpClientRequest> openUrl(String method, Uri url) {
+    final int id = count++;
+
+    // #1 • GET • https://flutter.io open
+    _log.log(() => '#$id \u2022 $method \u2022 $url open');
```

Comment:
I think we can just inline unicode bullets (•) instead of using the unicode escapes.